### PR TITLE
feat: email verification + company association

### PR DESCRIPTION
**Pull request by FFawzy**
This PR covers the following problems and partially addresses commercetools/learning-tech#454 and part of commercetools/learning-tech#429
- Allowing a user to see their email that they logged in with.
- Allowing the user to see whether their account is verified or not and if not to resend a verification email.
- Allowing the user to see whether their company association is verified or not and instructions on how to verify it.

Still missing
- do we need to show anything else to push for email verification ? a notification message/ banner because now it is hidden inside the profile modal only

Some Screenshots how it looks like

Screenshot 2023-08-16 at 12 26 02 pm
Screenshot 2023-08-16 at 12 26 47 pm

### CODE REVIEW

**Comment by zbalek** [comment]
Hi @FFawzy 

Re content, I'd add the word 'address' at the end of the sentence, i.e.:
_Please contact us at training@commercetools.com to change your email address._   

Also, can we perhaps make the email address clickable, opening an email client? Perhaps in that we could also add some instructions re how to verify the company associatio. I'm thinking we could pre-write the email when they open the email client, for example:

  Hi commercetools Training team,  
  
I'm writing to verify my company association.   Please see attached X.  
  
  Thanks,
  
  We could also include a subject in that email so that we can sort through those emails faster.
  
  ----------------------
  
  Re visual design: the 'Not verified' stamp is a little chunky and in this particular case, I'd opt for a stamo w/o an icon for non-verified but kept it for the verified stamp.

**Comment by FFawzy** [comment]
A couple of updates according to the comments of Nikolaus and Zoe


Screenshot 2023-08-17 at 4 09 05 pm
Screenshot 2023-08-17 at 4 13 19 pm


Trying the button next to the field as Nikolaus suggested (@nkuehn I don't think it looks good at all)
Screenshot 2023-08-17 at 3 53 13 pm


CC: @zbalek && @nkuehn

**Comment by nkuehn** [comment]
visual feedback:

image

> Trying the button next to the field as Nikolaus suggested
sorry, misunderstnading. I had considered the "verified" badge at that spot.  The button would best live below the text box

about the bottom text: 

image

**Comment by zbalek** [comment]
Re the text - "your company association is...". Is there any reason we are not using the "hint text" that is [available for fields in the UI kit? We could still kee

https://uikit.commercetools.com/?path=/story/components-fields--textfield&knob-name=&knob-hint=YourCompanyAssociationIs...&knob-hintIcon=&knob-horizontalConstraint=7&knob-errors={"missing":true,"customError":true}&knob-isRequired=&knob-touched=&knob-isAutofocussed=&knob-isDisabled=&knob-isReadOnly=&knob-placeholder=Placeholder&knob-title=Username&knob-description=&knob-show

--------------------------------------
**example** of the hint text
image

**Comment by FFawzy** [comment]
Screenshot 2023-08-18 at 10 52 47 am
Screenshot 2023-08-18 at 10 53 59 am

CC: @nkuehn && @zbalek

**Comment by zbalek** [comment]
@nkuehn I see what you mean. Why I recommended to use the space above the field is because whilst the text in this component is called 'hint', the information in the text is still related to the field below regardless of the actual content. However, if we choose to add text below the field, I'd recommend sticking to the same font as for the "hint", i.e. MC Rdesign Library - Details - color-neutral-40

image

**Comment by nkuehn** [comment]
> the information in the text is still related to the field below

It's not as directly related as it sounds (just because both are "company").  If we make the UI suggest that we verify what the user entered into the "company" field (we don't) instead of making transparent that we will maintain an independent separate "verified company" field we just have to pay for it in many emails and misunderstandings and personal explanations.

File: packages/gatsby-theme-docs/src/modules/self-learning/components/profile-modal.tsx [hunk 1]
```diff
@@ -148,20 +174,84 @@ const ProfileModal = () => {
...
+            errors={
+              TextField.toFieldErrors<TProfileFormValues>(formik.errors).company
+            }
+            renderError={renderError}
+            onChange={formik.handleChange}
+            onBlur={formik.handleBlur}
+          />
+          {profile?.global_account_name ? (
+            <Label>
+              Your company association is <b>{profile.global_account_name}</b>{' '}
```
Comment (nkuehn) [inline packages/gatsby-theme-docs/src/modules/self-learning/components/profile-modal.tsx:232]:
```suggestion
              Your verified company association is <b>{profile.global_account_name}</b>{' '}
```

**Review by nkuehn — APPROVED** [review]
Thank you!  

Please don't forget the enablement of the training operations team!

**Pull request merged**